src/main/java/ada/chore_api_v2/MissionChore/MissionChoreService.java

```diff
@@ -0,0 +1,56 @@
+package ada.chore_api_v2.MissionChore;
+
+import ada.chore_api_v2.Mission.Mission;
+import ada.chore_api_v2.Mission.MissionRepository;
+import ada.chore_api_v2.Chore.Chore;
+import ada.chore_api_v2.Chore.ChoreRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class MissionChoreService {
+
+    private final MissionChoreRepository missionChoreRepository;
+    private final ChoreRepository choreRepository;
+    private final MissionRepository missionRepository;
+
+    public MissionChoreService(MissionChoreRepository missionChoreRepository, ChoreRepository choreRepository, MissionRepository missionRepository) {
+        this.missionChoreRepository = missionChoreRepository;
+        this.choreRepository = choreRepository;
+        this.missionRepository = missionRepository;
+    }
+
+    //create new ChoreMission
+    public MissionChore createMissionChore( Integer missionId, Integer choreId, Boolean isCompleted) {
+        Optional<Mission> missionOptional = missionRepository.findById(missionId);
+        Optional<Chore> choreOptional = choreRepository.findById(choreId);
+
+        if (missionOptional.isPresent() && choreOptional.isPresent()) {
+            Mission mission = missionOptional.get();
+            Chore chore = choreOptional.get();
+
+            MissionChore missionChore = new MissionChore(mission, chore, isCompleted);
+            return missionChoreRepository.save(missionChore);
+        }
+        return null;
+    }
+
+
+    // get all ChoreMissions
+    public Iterable<MissionChore> getAllMissionChore()  {
+        return missionChoreRepository.findAll();
+    }
+
+    // update ChoreMission
+    public MissionChore updateMissionChore(MissionChoreId missionChoreId, Boolean isCompleted) {
+        Optional<MissionChore> missionChoreOptional = missionChoreRepository.findById(missionChoreId);
```

Comment:
Instead of making a new MissionChoreId in the controller, you should be able to make a method in the missionChorerepository called findByMissionIdAndChoreId(). 
```suggestion
        Optional<MissionChore> missionChoreOptional = missionChoreRepository.findByMissionIdAndChoreID(missionId, choreId);
```